pkg/services/sqlstore/user_auth.go

```diff
@@ -115,6 +116,30 @@ func GetUserByAuthInfo(query *m.GetUserByAuthInfoQuery) error {
 	return nil
 }
 
+func GetExternalUserInfoByLogin(query *m.GetExternalUserInfoByLoginQuery) error {
+	userQuery := m.GetUserByLoginQuery{LoginOrEmail: query.LoginOrEmail}
+	err := bus.Dispatch(&userQuery)
+	if err != nil {
+		return err
+	}
+
+	authInfoQuery := &m.GetAuthInfoQuery{UserId: userQuery.Result.Id}
```

Comment:
So that's two sql queries to get the user, right?

Reply:
Yes, the `GetAuthInfoQuery` query takes only user id, but we have user login only. So we need to get user by login first.

Reply:
I suppose underlying issue here is that we have an `user` table and supplementary `user_auth` table and the means we have right now (specifically API of sqlstore module) does not provide an sql request which would get us the needed info in one go?

If so, I would suggest an optimisation for later stages (or should we consider it for right now?) - we should extend sqlstore so it would return values of the `user_auth` too, that way we would decrease amount of requests and simplify the code abstracting it in sqlstore module.

It's doesn't really matter here, since it seems purpose of this code is to disable a single user? That shouldn't be a hot path.

But when we implement bulk changes - things probably will get nasty :)

Reply:
Making it into one query would be preferable but I think we can chage if, if needed, when we do the performance testing.

Reply:
Hm, I was thinking to do a perf test only for getting users into grafana - will take a note of that

Reply:
It's the update/sync that needs testing. Initial add of users to grafana will happen as they login and shouldn't be a performance issue as it only syncs one user at a time.